docs/spec.openapi.yaml

```diff
@@ -0,0 +1,627 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '401':
+          description: Not authorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /files/{fileId}:
+    delete:
+      tags:
+        - Files
+      description: Deletes an existing file
+      parameters:
+        - in: path
+          name: fileId
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: File successfully deleted
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '401':
+          description: Not authorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description:  Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /files/{fileId}/move:
+    patch:
+      tags:
+        - Files
+      description: Move an existing file to a different location
+      parameters:
+        - in: path
+          name: fileId
+          required: true
+          schema:
+            type: string
+        - in: query
+          name: newLocation
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: File moved successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '401':
+          description: Not authorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+                
+                
+  /files/list:
+    get:
+      tags:
+        - Files
+      description: List files in a given location
+      parameters:
+        - in: query
+          name: location
+          required: true
+          schema:
+            type: string  
+          description: The location (UUID) to list files from
+      responses:
+        '200':
+          description: List of files in the location
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/fileDetails'  
```

Comment:
According to the [File.java](https://github.com/hawks-atlanta/gateway-java/blob/dev/app/src/main/java/gateway/soap/response/File.java) response, this endpoint also returns the `UUID` and `isFile` attribute.